packages/mobile/src/utils/challenges.tsx

```diff
@@ -0,0 +1,252 @@
+import {
+  ChallengeRewardID,
+  TrendingRewardID
+} from 'audius-client/src/common/models/AudioRewards'
+import {
+  ACCOUNT_VERIFICATION_SETTINGS_PAGE,
+  TRENDING_PAGE,
+  UPLOAD_PAGE
+} from 'audius-client/src/utils/route'
+import { ImageSourcePropType } from 'react-native'
+
+import ChartIncreasing from 'app/assets/images/emojis/chart-increasing.png'
+import Headphone from 'app/assets/images/emojis/headphone.png'
+import IncomingEnvelope from 'app/assets/images/emojis/incoming-envelope.png'
+import LoveLetter from 'app/assets/images/emojis/love-letter.png'
+import MobilePhoneWithArrow from 'app/assets/images/emojis/mobile-phone-with-arrow.png'
+import MultipleMusicalNotes from 'app/assets/images/emojis/multiple-musical-notes.png'
+import NerdFace from 'app/assets/images/emojis/nerd-face.png'
+import WhiteHeavyCheckMark from 'app/assets/images/emojis/white-heavy-check-mark.png'
+import IconArrow from 'app/assets/images/iconArrow.svg'
+import IconCheck from 'app/assets/images/iconCheck.svg'
+import IconUpload from 'app/assets/images/iconUpload.svg'
+
+// TODO: These should be programmatic based on amount, but historically have not been
```

Comment:
The `description` (and `fullDescription`) is programmatic based on amount on web - we should pull that in here as well too. Mobile didn't have it prior for the drawers and I think we can probably leave it that way for now since it uses `fullDescription` which hardly references amount and the actual amount is used in the same modal, but since the audio page uses `description` and is the first thing seen I think it'd be good to make sure it's always accurate

Reply:
I think this requires a bit heavier lift b/c we currently include those descriptions in the objects that contain the web icons. I want to make a separate change where we consolidate all of these into one shared object.

Reply:
Yeah def will take more work to consolidate - I was thinking literal copy/paste to use functions for description/fullDescription instead of importing for now until we break out into a consolidated approach?

Reply:
Yeah that's fair. I can submit a subsequent PR. I just figured that because drawers already had hard-coded amts on mobile, this was a ~wash

Reply:
Yeah, true. I think they have the benefit of having the true amount in big purple letters right next to it, so even if the text gets out of date it's pretty obvious what it is. Also I think every challenge uses the amount in the normal description vs the full description only mentions it for a couple